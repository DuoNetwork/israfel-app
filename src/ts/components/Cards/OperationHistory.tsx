import { Popconfirm } from 'antd';
import * as React from 'react';
import wsUtil from 'ts/common/wsUtil';
import { IUserOrder } from '../../../../../israfel-relayer/src/common/types';
import * as CST from '../../common/constants';
import util from '../../common/util';
import { SDivFlexCenter } from '../_styled';
import { SCard, SCardTitle } from './_styled';
import { SCardList } from './_styled';

interface IProps {
	orderHistory: IUserOrder[];
	userOrder: IUserOrder[];
	locale: string;
}

interface IState {
	orderHash: string;
}

let orderHash: string = '';

export default class TimeSeriesCard extends React.Component<IProps, IState> {
	constructor(props: IProps) {
		super(props);
		this.state = {
			orderHash: ''
		};
	}

	private setOrderHash(index: any) {
		orderHash = this.props.orderHistory[index].orderHash;
	}

	private deleteOrder() {
		wsUtil.deleteOrder(orderHash);
	}
	public render() {
		let { orderHistory } = this.props;
		const { userOrder } = this.props;
		console.log(userOrder);
		for (let i = 0; i < userOrder.length; i++)
			if (userOrder[i].type === 'add')
				orderHistory = util.addOrder(orderHistory, userOrder[i]);
		orderHistory = userOrder;
		const title = CST.TH_ORDERBOOK.toUpperCase();
		const step = orderHistory ? util.range(0, orderHistory.length) : [];
		return (
			<SCard title={<SCardTitle>{title}</SCardTitle>} width="800px" margin="0 10px 0 0">
				<SDivFlexCenter center horizontal>
					<SCardList>
						<div className="status-list-wrapper">
							<ul>
								<li className="right">
									<span className="title">{CST.TH_ACTIONS.toUpperCase()}</span>
								</li>
								{orderHistory && orderHistory.length ? (
									util.range(0, orderHistory.length).map((i: any) => (
										<li key={i} style={{ height: '28px' }}>
											<span className="content">
												{i < orderHistory.length
													? orderHistory[i].amount !== 0
														? util.formatNumber(orderHistory[i].amount)
														: '-'
													: '-'}
											</span>
											<span className="title">
												{i < orderHistory.length
													? orderHistory[i].price !== 0
														? util.formatNumber(orderHistory[i].price)
														: '-'
													: '-'}
											</span>
											<span className="title">
												{i < orderHistory.length
													? orderHistory[i].price !== 0
														? orderHistory[i].side
														: '-'
													: '-'}
											</span>
										</li>
									))
								) : (
									<li className="block-title t-center">{CST.TH_LOADING}</li>
								)}
							</ul>
						</div>
					</SCardList>

					<SCardList fixWidth style={{ width: '100px' }}>
						<div className="status-list-wrapper narrow">
							<ul className="noborder">
								<li>
									<span className="title">{CST.TH_STATUS}</span>
								</li>
								{step.length > 0 ? (
									step.map((i: any) => (
										<li key={i} style={{ height: '28px' }}>
											<span className="title">{orderHistory[i].status}</span>
										</li>
									))
								) : (
									<li className="block-title t-center">...</li>
								)}
							</ul>
						</div>
					</SCardList>

					<SCardList fixWidth style={{ width: '150px' }}>
						<div className="status-list-wrapper narrow">
							<ul className="noborder">
								<li>
									<span className="title">{CST.TH_ACTIONS}</span>
								</li>
								{step.length > 0 ? (
									step.map((i: any) => (
										<Popconfirm
											placement="top"
											title={CST.TH_DELETE_ORDER}
											okText="Confirm"
											cancelText="Cancel"
											onConfirm={this.deleteOrder}
										>
											<li
												key={i}
												style={{ height: '28px' }}
												onClick={this.setOrderHash.bind(this, i)}
											>
												<button
													className={'form-button'}
													disabled={orderHistory[i].status === 'pending'}
												>
													cancel
												</button>
											</li>
										</Popconfirm>
									))
								) : (
									<li className="block-title t-center">...</li>
								)}
							</ul>
						</div>
					</SCardList>
				</SDivFlexCenter>
			</SCard>
		);
	}
}
