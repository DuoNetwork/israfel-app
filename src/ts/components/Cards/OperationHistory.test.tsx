import { Popconfirm } from 'antd';
// import { Select } from 'antd';
import { shallow } from 'enzyme';
import * as React from 'react';
import { SCard } from './_styled';
import OperationHistory from './OperationHistory';

describe('OperationHistory Test', () => {
	const orderHistory = [
		{
			account: '0x66ad9d0b933da88bbee196b2a9c0badc901c4a3a',
			amount: 2,
			balance: 1,
			fill: 0,
			createdAt: 1541143007295,
			currentSequence: 0,
			initialSequence: 0,
			orderHash: '0xc3a8341c6d29bf266943b4822b4be08a2aac7a1bf820c26737147dac0428ac6f',
			pair: 'ZRX-WETH',
			price: 0.16666666,
			side: 'ask',
			status: 'confirm',
			type: 'add',
			updatedAt: 1541143007295,
			updatedBy: 'user'
		},
		{
			account: '0x66ad9d0b933da88bbee196b2a9c0badc901c4a3a',
			amount: 2,
			balance: 1,
			fill: 0,
			createdAt: 1541143007295,
			currentSequence: 0,
			initialSequence: 0,
			orderHash: '0xc3a8341c6e29bf266943b4822b4be08a2aac7a1bf820c26737147dac0428ac6f',
			pair: 'ZRX-WETH',
			price: 0.16666666,
			side: 'ask',
			status: 'pending',
			type: 'add',
			updatedAt: 1541143007296,
			updatedBy: 'user'
		}
	];
	describe('User Login', () => {
		it('Test Snapshot', () => {
			const wrapper = shallow(
				<OperationHistory
					locale={''}
					orderHistory={orderHistory}
					userOrder={orderHistory}
				/>
			);
			expect(wrapper).toMatchSnapshot();
			wrapper.find(SCard)
			// .find(SCardTitle)
			// .find(Select)
			.simulate('change', {target: { value: "Summary" }});
			expect(wrapper).toMatchSnapshot();
			wrapper.find(Popconfirm).at(0).simulate('confirm');
			expect(wrapper).toMatchSnapshot();
			wrapper.find('li').at(6).simulate('click');
			expect(wrapper).toMatchSnapshot();
		});
	});
});
