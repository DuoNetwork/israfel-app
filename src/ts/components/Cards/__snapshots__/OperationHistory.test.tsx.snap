// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperationHistory Test User Login Test Snapshot 1`] = `
<Styled(Card)
  margin="0 10px 0 0"
  title={
    <styled.div>
      ORDER HISTORY
      <Select
        choiceTransitionName="zoom"
        onChange={[Function]}
        prefixCls="ant-select"
        showSearch={false}
        style={
          Object {
            "width": 200,
          }
        }
        transitionName="slide-up"
        value="Summary"
      >
        <Option
          value="Detail"
        >
          Detail
        </Option>
        <Option
          value="Summary"
        >
          Summary
        </Option>
      </Select>
    </styled.div>
  }
  width="800px"
>
  <styled.div
    center={true}
    horizontal={true}
  >
    <styled.div>
      <div
        className="status-list-wrapper"
      >
        <ul>
          <li
            className="right"
          >
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              AMT
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              BAL
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              FILL
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              PRICE
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              SIDE
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              Actions
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              Time
            </span>
          </li>
          <li
            key="0"
            style={
              Object {
                "height": "40px",
              }
            }
          >
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              2.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              1.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              -
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              0.1667
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              ask
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              add
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              02-11-2018 15:16:47
            </span>
          </li>
          <li
            key="1"
            style={
              Object {
                "height": "40px",
              }
            }
          >
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              2.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              1.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              -
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              0.1667
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              ask
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              add
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              02-11-2018 15:16:47
            </span>
          </li>
        </ul>
      </div>
    </styled.div>
    <styled.div
      fixWidth={true}
      style={
        Object {
          "width": "100px",
        }
      }
    >
      <div
        className="status-list-wrapper narrow"
      >
        <ul
          className="noborder"
        >
          <li>
            <span
              className="title"
            >
              Status
            </span>
          </li>
          <li
            key="0"
            style={
              Object {
                "height": "28px",
              }
            }
          >
            <span
              className="title"
            >
              confirm
            </span>
          </li>
          <li
            key="1"
            style={
              Object {
                "height": "28px",
              }
            }
          >
            <span
              className="title"
            >
              pending
            </span>
          </li>
        </ul>
      </div>
    </styled.div>
    <styled.div
      fixWidth={true}
      style={
        Object {
          "width": "150px",
        }
      }
    >
      <div
        className="status-list-wrapper narrow"
      >
        <ul
          className="noborder"
        >
          <li>
            <span
              className="title"
            >
              Actions
            </span>
          </li>
          <Popconfirm
            cancelText="Cancel"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            key="0"
            okText="Confirm"
            okType="primary"
            onConfirm={[Function]}
            placement="top"
            prefixCls="ant-popover"
            title="Are you sure to delete this order?"
            transitionName="zoom-big"
            trigger="click"
          >
            <li
              key="0"
              onClick={[Function]}
              style={
                Object {
                  "height": "28px",
                }
              }
            >
              <button
                className="form-button"
                disabled={false}
              >
                cancel
              </button>
            </li>
          </Popconfirm>
          <Popconfirm
            cancelText="Cancel"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            key="1"
            okText="Confirm"
            okType="primary"
            onConfirm={[Function]}
            placement="top"
            prefixCls="ant-popover"
            title="Are you sure to delete this order?"
            transitionName="zoom-big"
            trigger="click"
          >
            <li
              key="1"
              onClick={[Function]}
              style={
                Object {
                  "height": "28px",
                }
              }
            >
              <button
                className="form-button"
                disabled={true}
              >
                cancel
              </button>
            </li>
          </Popconfirm>
        </ul>
      </div>
    </styled.div>
  </styled.div>
</Styled(Card)>
`;

exports[`OperationHistory Test User Login Test Snapshot 2`] = `
<Styled(Card)
  margin="0 10px 0 0"
  title={
    <styled.div>
      ORDER HISTORY
      <Select
        choiceTransitionName="zoom"
        onChange={[Function]}
        prefixCls="ant-select"
        showSearch={false}
        style={
          Object {
            "width": 200,
          }
        }
        transitionName="slide-up"
        value="Summary"
      >
        <Option
          value="Detail"
        >
          Detail
        </Option>
        <Option
          value="Summary"
        >
          Summary
        </Option>
      </Select>
    </styled.div>
  }
  width="800px"
>
  <styled.div
    center={true}
    horizontal={true}
  >
    <styled.div>
      <div
        className="status-list-wrapper"
      >
        <ul>
          <li
            className="right"
          >
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              AMT
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              BAL
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              FILL
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              PRICE
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              SIDE
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              Actions
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              Time
            </span>
          </li>
          <li
            key="0"
            style={
              Object {
                "height": "40px",
              }
            }
          >
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              2.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              1.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              -
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              0.1667
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              ask
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              add
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              02-11-2018 15:16:47
            </span>
          </li>
          <li
            key="1"
            style={
              Object {
                "height": "40px",
              }
            }
          >
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              2.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              1.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              -
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              0.1667
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              ask
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              add
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              02-11-2018 15:16:47
            </span>
          </li>
        </ul>
      </div>
    </styled.div>
    <styled.div
      fixWidth={true}
      style={
        Object {
          "width": "100px",
        }
      }
    >
      <div
        className="status-list-wrapper narrow"
      >
        <ul
          className="noborder"
        >
          <li>
            <span
              className="title"
            >
              Status
            </span>
          </li>
          <li
            key="0"
            style={
              Object {
                "height": "28px",
              }
            }
          >
            <span
              className="title"
            >
              confirm
            </span>
          </li>
          <li
            key="1"
            style={
              Object {
                "height": "28px",
              }
            }
          >
            <span
              className="title"
            >
              pending
            </span>
          </li>
        </ul>
      </div>
    </styled.div>
    <styled.div
      fixWidth={true}
      style={
        Object {
          "width": "150px",
        }
      }
    >
      <div
        className="status-list-wrapper narrow"
      >
        <ul
          className="noborder"
        >
          <li>
            <span
              className="title"
            >
              Actions
            </span>
          </li>
          <Popconfirm
            cancelText="Cancel"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            key="0"
            okText="Confirm"
            okType="primary"
            onConfirm={[Function]}
            placement="top"
            prefixCls="ant-popover"
            title="Are you sure to delete this order?"
            transitionName="zoom-big"
            trigger="click"
          >
            <li
              key="0"
              onClick={[Function]}
              style={
                Object {
                  "height": "28px",
                }
              }
            >
              <button
                className="form-button"
                disabled={false}
              >
                cancel
              </button>
            </li>
          </Popconfirm>
          <Popconfirm
            cancelText="Cancel"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            key="1"
            okText="Confirm"
            okType="primary"
            onConfirm={[Function]}
            placement="top"
            prefixCls="ant-popover"
            title="Are you sure to delete this order?"
            transitionName="zoom-big"
            trigger="click"
          >
            <li
              key="1"
              onClick={[Function]}
              style={
                Object {
                  "height": "28px",
                }
              }
            >
              <button
                className="form-button"
                disabled={true}
              >
                cancel
              </button>
            </li>
          </Popconfirm>
        </ul>
      </div>
    </styled.div>
  </styled.div>
</Styled(Card)>
`;

exports[`OperationHistory Test User Login Test Snapshot 3`] = `
<Styled(Card)
  margin="0 10px 0 0"
  title={
    <styled.div>
      ORDER HISTORY
      <Select
        choiceTransitionName="zoom"
        onChange={[Function]}
        prefixCls="ant-select"
        showSearch={false}
        style={
          Object {
            "width": 200,
          }
        }
        transitionName="slide-up"
        value="Summary"
      >
        <Option
          value="Detail"
        >
          Detail
        </Option>
        <Option
          value="Summary"
        >
          Summary
        </Option>
      </Select>
    </styled.div>
  }
  width="800px"
>
  <styled.div
    center={true}
    horizontal={true}
  >
    <styled.div>
      <div
        className="status-list-wrapper"
      >
        <ul>
          <li
            className="right"
          >
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              AMT
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              BAL
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              FILL
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              PRICE
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              SIDE
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              Actions
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              Time
            </span>
          </li>
          <li
            key="0"
            style={
              Object {
                "height": "40px",
              }
            }
          >
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              2.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              1.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              -
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              0.1667
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              ask
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              add
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              02-11-2018 15:16:47
            </span>
          </li>
          <li
            key="1"
            style={
              Object {
                "height": "40px",
              }
            }
          >
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              2.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              1.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              -
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              0.1667
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              ask
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              add
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              02-11-2018 15:16:47
            </span>
          </li>
        </ul>
      </div>
    </styled.div>
    <styled.div
      fixWidth={true}
      style={
        Object {
          "width": "100px",
        }
      }
    >
      <div
        className="status-list-wrapper narrow"
      >
        <ul
          className="noborder"
        >
          <li>
            <span
              className="title"
            >
              Status
            </span>
          </li>
          <li
            key="0"
            style={
              Object {
                "height": "28px",
              }
            }
          >
            <span
              className="title"
            >
              confirm
            </span>
          </li>
          <li
            key="1"
            style={
              Object {
                "height": "28px",
              }
            }
          >
            <span
              className="title"
            >
              pending
            </span>
          </li>
        </ul>
      </div>
    </styled.div>
    <styled.div
      fixWidth={true}
      style={
        Object {
          "width": "150px",
        }
      }
    >
      <div
        className="status-list-wrapper narrow"
      >
        <ul
          className="noborder"
        >
          <li>
            <span
              className="title"
            >
              Actions
            </span>
          </li>
          <Popconfirm
            cancelText="Cancel"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            key="0"
            okText="Confirm"
            okType="primary"
            onConfirm={[Function]}
            placement="top"
            prefixCls="ant-popover"
            title="Are you sure to delete this order?"
            transitionName="zoom-big"
            trigger="click"
          >
            <li
              key="0"
              onClick={[Function]}
              style={
                Object {
                  "height": "28px",
                }
              }
            >
              <button
                className="form-button"
                disabled={false}
              >
                cancel
              </button>
            </li>
          </Popconfirm>
          <Popconfirm
            cancelText="Cancel"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            key="1"
            okText="Confirm"
            okType="primary"
            onConfirm={[Function]}
            placement="top"
            prefixCls="ant-popover"
            title="Are you sure to delete this order?"
            transitionName="zoom-big"
            trigger="click"
          >
            <li
              key="1"
              onClick={[Function]}
              style={
                Object {
                  "height": "28px",
                }
              }
            >
              <button
                className="form-button"
                disabled={true}
              >
                cancel
              </button>
            </li>
          </Popconfirm>
        </ul>
      </div>
    </styled.div>
  </styled.div>
</Styled(Card)>
`;

exports[`OperationHistory Test User Login Test Snapshot 4`] = `
<Styled(Card)
  margin="0 10px 0 0"
  title={
    <styled.div>
      ORDER HISTORY
      <Select
        choiceTransitionName="zoom"
        onChange={[Function]}
        prefixCls="ant-select"
        showSearch={false}
        style={
          Object {
            "width": 200,
          }
        }
        transitionName="slide-up"
        value="Summary"
      >
        <Option
          value="Detail"
        >
          Detail
        </Option>
        <Option
          value="Summary"
        >
          Summary
        </Option>
      </Select>
    </styled.div>
  }
  width="800px"
>
  <styled.div
    center={true}
    horizontal={true}
  >
    <styled.div>
      <div
        className="status-list-wrapper"
      >
        <ul>
          <li
            className="right"
          >
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              AMT
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              BAL
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              FILL
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              PRICE
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              SIDE
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              Actions
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              Time
            </span>
          </li>
          <li
            key="0"
            style={
              Object {
                "height": "40px",
              }
            }
          >
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              2.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              1.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              -
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              0.1667
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              ask
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              add
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              02-11-2018 15:16:47
            </span>
          </li>
          <li
            key="1"
            style={
              Object {
                "height": "40px",
              }
            }
          >
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              2.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              1.00
            </span>
            <span
              className="content"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              -
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              0.1667
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              ask
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 30,
                }
              }
            >
              add
            </span>
            <span
              className="title"
              style={
                Object {
                  "textAlign": "center",
                  "width": 120,
                }
              }
            >
              02-11-2018 15:16:47
            </span>
          </li>
        </ul>
      </div>
    </styled.div>
    <styled.div
      fixWidth={true}
      style={
        Object {
          "width": "100px",
        }
      }
    >
      <div
        className="status-list-wrapper narrow"
      >
        <ul
          className="noborder"
        >
          <li>
            <span
              className="title"
            >
              Status
            </span>
          </li>
          <li
            key="0"
            style={
              Object {
                "height": "28px",
              }
            }
          >
            <span
              className="title"
            >
              confirm
            </span>
          </li>
          <li
            key="1"
            style={
              Object {
                "height": "28px",
              }
            }
          >
            <span
              className="title"
            >
              pending
            </span>
          </li>
        </ul>
      </div>
    </styled.div>
    <styled.div
      fixWidth={true}
      style={
        Object {
          "width": "150px",
        }
      }
    >
      <div
        className="status-list-wrapper narrow"
      >
        <ul
          className="noborder"
        >
          <li>
            <span
              className="title"
            >
              Actions
            </span>
          </li>
          <Popconfirm
            cancelText="Cancel"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            key="0"
            okText="Confirm"
            okType="primary"
            onConfirm={[Function]}
            placement="top"
            prefixCls="ant-popover"
            title="Are you sure to delete this order?"
            transitionName="zoom-big"
            trigger="click"
          >
            <li
              key="0"
              onClick={[Function]}
              style={
                Object {
                  "height": "28px",
                }
              }
            >
              <button
                className="form-button"
                disabled={false}
              >
                cancel
              </button>
            </li>
          </Popconfirm>
          <Popconfirm
            cancelText="Cancel"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            key="1"
            okText="Confirm"
            okType="primary"
            onConfirm={[Function]}
            placement="top"
            prefixCls="ant-popover"
            title="Are you sure to delete this order?"
            transitionName="zoom-big"
            trigger="click"
          >
            <li
              key="1"
              onClick={[Function]}
              style={
                Object {
                  "height": "28px",
                }
              }
            >
              <button
                className="form-button"
                disabled={true}
              >
                cancel
              </button>
            </li>
          </Popconfirm>
        </ul>
      </div>
    </styled.div>
  </styled.div>
</Styled(Card)>
`;
